File: Devart/dotConnect/PostgreSQL/Samples/General/VB/StoredProc/StoredProcDemoControl.vb
Imports Devart.Data.PostgreSql
Imports System
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Windows.Forms

Namespace Samples
  Public Class StoredProcDemoControl
    Inherits BaseDemoControl

    ' Fields
    Private WithEvents topPanel As System.Windows.Forms.Panel
    Private WithEvents lbParams As System.Windows.Forms.Label
    Private WithEvents tbLoc As System.Windows.Forms.TextBox
    Private WithEvents tbDname As System.Windows.Forms.TextBox
    Private WithEvents tbDeptno As System.Windows.Forms.TextBox
    Private WithEvents lbParam3 As System.Windows.Forms.Label
    Private WithEvents lbParam2 As System.Windows.Forms.Label
    Private WithEvents lbParam1 As System.Windows.Forms.Label
    Private WithEvents btClear As System.Windows.Forms.Button
    Private WithEvents btFill As System.Windows.Forms.Button
    Private WithEvents btExecute As System.Windows.Forms.Button
    Private WithEvents edStoredProcName As System.Windows.Forms.ComboBox
    Private WithEvents lbStoredProc As System.Windows.Forms.Label
    Private WithEvents procedureCommand As Devart.Data.PostgreSql.PgSqlCommand
    Private WithEvents selectCommand As Devart.Data.PostgreSql.PgSqlCommand
    Private WithEvents dataAdapter As Devart.Data.PostgreSql.PgSqlDataAdapter
    Private WithEvents dataGrid As System.Windows.Forms.DataGrid

    ' Methods
    Public Sub New()
      Me.components = Nothing
      Me.InitializeComponent()
    End Sub

    Public Sub New(ByVal connection As PgSqlConnection)
      Me.New()
      Dim locConnection As PgSqlConnection = connection
      Me.procedureCommand.Connection = locConnection
      Me.selectCommand.Connection = locConnection
      Me.edStoredProcName.Text = Me.procedureCommand.CommandText
      Me.tbDeptno.Text = Me.procedureCommand.Parameters.Item("@pdeptno").Value.ToString
      Me.tbDname.Text = Me.procedureCommand.Parameters.Item("@pdname").Value.ToString
      Me.tbLoc.Text = Me.procedureCommand.Parameters.Item("@ploc").Value.ToString
    End Sub

    'UserControl overrides dispose to clean up the component list.
    Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
      If disposing Then
        If Not (components Is Nothing) Then
          components.Dispose()
        End If
      End If
      MyBase.Dispose(disposing)
    End Sub

    'Required by the Windows Form Designer
    Private components As System.ComponentModel.IContainer

    'NOTE: The following procedure is required by the Windows Form Designer
    'It can be modified using the Windows Form Designer.  
    'Do not modify it using the code editor.
    <System.Diagnostics.DebuggerStepThrough()> _
    Private Sub InitializeComponent()
      Me.topPanel = New System.Windows.Forms.Panel()
      Me.lbParams = New System.Windows.Forms.Label()
      Me.tbLoc = New System.Windows.Forms.TextBox()
      Me.tbDname = New System.Windows.Forms.TextBox()
      Me.tbDeptno = New System.Windows.Forms.TextBox()
      Me.lbParam3 = New System.Windows.Forms.Label()
      Me.lbParam2 = New System.Windows.Forms.Label()
      Me.lbParam1 = New System.Windows.Forms.Label()
      Me.btClear = New System.Windows.Forms.Button()
      Me.btFill = New System.Windows.Forms.Button()
      Me.btExecute = New System.Windows.Forms.Button()
      Me.edStoredProcName = New System.Windows.Forms.ComboBox()
      Me.lbStoredProc = New System.Windows.Forms.Label()
      Me.dataGrid = New System.Windows.Forms.DataGrid()
      Me.procedureCommand = New Devart.Data.PostgreSql.PgSqlCommand()
      Me.selectCommand = New Devart.Data.PostgreSql.PgSqlCommand()
      Me.dataAdapter = New Devart.Data.PostgreSql.PgSqlDataAdapter()
      Me.topPanel.SuspendLayout()
      CType(Me.dataGrid, System.ComponentModel.ISupportInitialize).BeginInit()
      Me.SuspendLayout()
      '
      'topPanel
      '
      Me.topPanel.Controls.Add(Me.lbParams)
      Me.topPanel.Controls.Add(Me.tbLoc)
      Me.topPanel.Controls.Add(Me.tbDname)
      Me.topPanel.Controls.Add(Me.tbDeptno)
      Me.topPanel.Controls.Add(Me.lbParam3)
      Me.topPanel.Controls.Add(Me.lbParam2)
      Me.topPanel.Controls.Add(Me.lbParam1)
      Me.topPanel.Controls.Add(Me.btClear)
      Me.topPanel.Controls.Add(Me.btFill)
      Me.topPanel.Controls.Add(Me.btExecute)
      Me.topPanel.Controls.Add(Me.edStoredProcName)
      Me.topPanel.Controls.Add(Me.lbStoredProc)
      Me.topPanel.Dock = System.Windows.Forms.DockStyle.Top
      Me.topPanel.Location = New System.Drawing.Point(0, 0)
      Me.topPanel.Name = "topPanel"
      Me.topPanel.Size = New System.Drawing.Size(477, 88)
      Me.topPanel.TabIndex = 9
      '
      'lbParams
      '
      Me.lbParams.AutoSize = True
      Me.lbParams.Location = New System.Drawing.Point(8, 32)
      Me.lbParams.Name = "lbParams"
      Me.lbParams.Size = New System.Drawing.Size(147, 13)
      Me.lbParams.TabIndex = 18
      Me.lbParams.Text = "Stored procedure parameters:"
      '
      'tbLoc
      '
      Me.tbLoc.Location = New System.Drawing.Point(338, 52)
      Me.tbLoc.Name = "tbLoc"
      Me.tbLoc.Size = New System.Drawing.Size(127, 20)
      Me.tbLoc.TabIndex = 17
      '
      'tbDname
      '
      Me.tbDname.Location = New System.Drawing.Point(171, 52)
      Me.tbDname.Name = "tbDname"
      Me.tbDname.Size = New System.Drawing.Size(119, 20)
      Me.tbDname.TabIndex = 16
      '
      'tbDeptno
      '
      Me.tbDeptno.Location = New System.Drawing.Point(47, 52)
      Me.tbDeptno.Name = "tbDeptno"
      Me.tbDeptno.Size = New System.Drawing.Size(65, 20)
      Me.tbDeptno.TabIndex = 15
      '
      'lbParam3
      '
      Me.lbParam3.AutoSize = True
      Me.lbParam3.Location = New System.Drawing.Point(311, 56)
      Me.lbParam3.Name = "lbParam3"
      Me.lbParam3.Size = New System.Drawing.Size(25, 13)
      Me.lbParam3.TabIndex = 14
      Me.lbParam3.Text = "Loc"
      '
      'lbParam2
      '
      Me.lbParam2.AutoSize = True
      Me.lbParam2.Location = New System.Drawing.Point(128, 55)
      Me.lbParam2.Name = "lbParam2"
      Me.lbParam2.Size = New System.Drawing.Size(41, 13)
      Me.lbParam2.TabIndex = 13
      Me.lbParam2.Text = "Dname"
      '
      'lbParam1
      '
      Me.lbParam1.AutoSize = True
      Me.lbParam1.Location = New System.Drawing.Point(3, 54)
      Me.lbParam1.Name = "lbParam1"
      Me.lbParam1.Size = New System.Drawing.Size(42, 13)
      Me.lbParam1.TabIndex = 12
      Me.lbParam1.Text = "Deptno"
      '
      'btClear
      '
      Me.btClear.FlatStyle = System.Windows.Forms.FlatStyle.System
      Me.btClear.Location = New System.Drawing.Point(152, 0)
      Me.btClear.Name = "btClear"
      Me.btClear.Size = New System.Drawing.Size(75, 23)
      Me.btClear.TabIndex = 11
      Me.btClear.Text = "Clear"
      '
      'btFill
      '
      Me.btFill.FlatStyle = System.Windows.Forms.FlatStyle.System
      Me.btFill.Location = New System.Drawing.Point(76, 0)
      Me.btFill.Name = "btFill"
      Me.btFill.Size = New System.Drawing.Size(75, 23)
      Me.btFill.TabIndex = 10
      Me.btFill.Text = "Fill"
      '
      'btExecute
      '
      Me.btExecute.FlatStyle = System.Windows.Forms.FlatStyle.System
      Me.btExecute.Location = New System.Drawing.Point(0, 0)
      Me.btExecute.Name = "btExecute"
      Me.btExecute.Size = New System.Drawing.Size(75, 23)
      Me.btExecute.TabIndex = 9
      Me.btExecute.Text = "Execute"
      '
      'edStoredProcName
      '
      Me.edStoredProcName.DropDownStyle = System.Windows.Forms.ComboBoxStyle.DropDownList
      Me.edStoredProcName.Items.AddRange(New Object() {"Dept_Insert", "Dept_Update"})
      Me.edStoredProcName.Location = New System.Drawing.Point(328, 6)
      Me.edStoredProcName.Name = "edStoredProcName"
      Me.edStoredProcName.Size = New System.Drawing.Size(144, 21)
      Me.edStoredProcName.TabIndex = 8
      '
      'lbStoredProc
      '
      Me.lbStoredProc.AutoSize = True
      Me.lbStoredProc.Location = New System.Drawing.Point(232, 8)
      Me.lbStoredProc.Name = "lbStoredProc"
      Me.lbStoredProc.Size = New System.Drawing.Size(89, 13)
      Me.lbStoredProc.TabIndex = 7
      Me.lbStoredProc.Text = "Stored procedure"
      '
      'dataGrid
      '
      Me.dataGrid.BorderStyle = System.Windows.Forms.BorderStyle.None
      Me.dataGrid.CaptionVisible = False
      Me.dataGrid.DataMember = ""
      Me.dataGrid.Dock = System.Windows.Forms.DockStyle.Fill
      Me.dataGrid.HeaderForeColor = System.Drawing.SystemColors.ControlText
      Me.dataGrid.Location = New System.Drawing.Point(0, 88)
      Me.dataGrid.Name = "dataGrid"
      Me.dataGrid.Size = New System.Drawing.Size(477, 246)
      Me.dataGrid.TabIndex = 11
      '
      'procedureCommand
      '
      Me.procedureCommand.CommandText = "Dept_Insert"
      Me.procedureCommand.CommandType = System.Data.CommandType.StoredProcedure
      Me.procedureCommand.Name = "procedureCommand"
      Me.procedureCommand.Parameters.Add(New Devart.Data.PostgreSql.PgSqlParameter("@pdeptno", Devart.Data.PostgreSql.PgSqlType.Int, 3, System.Data.ParameterDirection.Input, True, CType(0, Byte), CType(0, Byte), "", System.Data.DataRowVersion.Current, "50"))
      Me.procedureCommand.Parameters.Add(New Devart.Data.PostgreSql.PgSqlParameter("@pdname", Devart.Data.PostgreSql.PgSqlType.VarChar, 10, System.Data.ParameterDirection.Input, True, CType(0, Byte), CType(0, Byte), "", System.Data.DataRowVersion.Current, "RESEARCH"))
      Me.procedureCommand.Parameters.Add(New Devart.Data.PostgreSql.PgSqlParameter("@ploc", Devart.Data.PostgreSql.PgSqlType.VarChar, 10, System.Data.ParameterDirection.Input, True, CType(0, Byte), CType(0, Byte), "", System.Data.DataRowVersion.Current, "NEW YORK"))
      '
      'selectCommand
      '
      Me.selectCommand.CommandText = "SELECT * FROM DEPT"
      Me.selectCommand.Name = "selectCommand"
      '
      'dataAdapter
      '
      Me.dataAdapter.SelectCommand = Me.selectCommand
      '
      'StoredProcDemoControl
      '
      Me.Controls.Add(Me.dataGrid)
      Me.Controls.Add(Me.topPanel)
      Me.Name = "StoredProcDemoControl"
      Me.Size = New System.Drawing.Size(477, 334)
      Me.topPanel.ResumeLayout(False)
      Me.topPanel.PerformLayout()
      CType(Me.dataGrid, System.ComponentModel.ISupportInitialize).EndInit()
      Me.ResumeLayout(False)

    End Sub

    ' Methods
    Private Sub btExecute_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btExecute.Click
      Me.procedureCommand.CommandText = Me.edStoredProcName.Text
      Me.procedureCommand.Parameters.Item("@pdeptno").Value = Convert.ToInt32(Me.tbDeptno.Text)
      Me.procedureCommand.Parameters.Item("@pdname").Value = Me.tbDname.Text
      Me.procedureCommand.Parameters.Item("@ploc").Value = Me.tbLoc.Text
      Me.procedureCommand.ExecuteNonQuery()
      MyBase.fieldWriteStatus1 = "Procedure executed successfully"
      MyBase.OnWriteStatus()
    End Sub

    Private Sub btFill_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btFill.Click
      Dim ds As New DataSet()
      Me.dataAdapter.Fill(ds)
      Me.dataGrid.DataSource = ds.Tables.Item(0)
    End Sub

    Private Sub btClear_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btClear.Click
      Me.dataGrid.DataSource = Nothing
      MyBase.fieldWriteStatus1 = ""
      MyBase.OnWriteStatus()
    End Sub
  End Class
End Namespace

File: Devart/dotConnect/PostgreSQL/Samples/General/CS/StoredProc/StoredProcDemoControl.cs
using System;

namespace Samples {
  public class StoredProcDemoControl : BaseDemoControl {

    private System.Windows.Forms.DataGrid dataGrid;
    private System.Windows.Forms.Panel topPanel;
    private System.Windows.Forms.TextBox tbLoc;
    private System.Windows.Forms.TextBox tbDname;
    private System.Windows.Forms.TextBox tbDeptno;
    private System.Windows.Forms.Button btClear;
    private System.Windows.Forms.Button btFill;
    private System.Windows.Forms.Button btExecute;
    private System.Windows.Forms.ComboBox edStoredProcName;
    private System.Windows.Forms.Label lbStoredProc;
    private Devart.Data.PostgreSql.PgSqlCommand procedureCommand;
    private Devart.Data.PostgreSql.PgSqlCommand selectCommand;
    private Devart.Data.PostgreSql.PgSqlDataAdapter dataAdapter;
    private System.Windows.Forms.Label lbParams;
    private System.Windows.Forms.Label lbParam3;
    private System.Windows.Forms.Label lbParam2;
    private System.Windows.Forms.Label lbParam1;

    private System.ComponentModel.IContainer components = null;

    public StoredProcDemoControl() {

      InitializeComponent();
    }
    
    public StoredProcDemoControl(Devart.Data.PostgreSql.PgSqlConnection connection) 
      : this () {

      Devart.Data.PostgreSql.PgSqlConnection locConnection = connection;
      procedureCommand.Connection = locConnection;
      selectCommand.Connection = locConnection;
      edStoredProcName.Text = procedureCommand.CommandText;
      tbDeptno.Text = procedureCommand.Parameters["@pdeptno"].Value.ToString();
      tbDname.Text = procedureCommand.Parameters["@pdname"].Value.ToString();
      tbLoc.Text = procedureCommand.Parameters["@ploc"].Value.ToString();
    }

    /// <summary>
    /// Clean up any resources being used.
    /// </summary>
    protected override void Dispose( bool disposing ) {
      if( disposing ) {
        if (components != null) {
          components.Dispose();
        }
      }
      base.Dispose( disposing );
    }

    #region Designer generated code
    /// <summary>
    /// Required method for Designer support - do not modify
    /// the contents of this method with the code editor.
    /// </summary>
    private void InitializeComponent() {
      this.procedureCommand = new Devart.Data.PostgreSql.PgSqlCommand();
      this.selectCommand = new Devart.Data.PostgreSql.PgSqlCommand();
      this.dataAdapter = new Devart.Data.PostgreSql.PgSqlDataAdapter();
      this.dataGrid = new System.Windows.Forms.DataGrid();
      this.topPanel = new System.Windows.Forms.Panel();
      this.lbParams = new System.Windows.Forms.Label();
      this.tbLoc = new System.Windows.Forms.TextBox();
      this.tbDname = new System.Windows.Forms.TextBox();
      this.tbDeptno = new System.Windows.Forms.TextBox();
      this.lbParam3 = new System.Windows.Forms.Label();
      this.lbParam2 = new System.Windows.Forms.Label();
      this.lbParam1 = new System.Windows.Forms.Label();
      this.btClear = new System.Windows.Forms.Button();
      this.btFill = new System.Windows.Forms.Button();
      this.btExecute = new System.Windows.Forms.Button();
      this.edStoredProcName = new System.Windows.Forms.ComboBox();
      this.lbStoredProc = new System.Windows.Forms.Label();
      ((System.ComponentModel.ISupportInitialize)(this.dataGrid)).BeginInit();
      this.topPanel.SuspendLayout();
      this.SuspendLayout();
      // 
      // procedureCommand
      // 
      this.procedureCommand.CommandType = System.Data.CommandType.StoredProcedure;
      this.procedureCommand.CommandText = "Dept_Insert";
      this.procedureCommand.Name = "procedureCommand";
      this.procedureCommand.Parameters.Add(new Devart.Data.PostgreSql.PgSqlParameter("@pdeptno", Devart.Data.PostgreSql.PgSqlType.Int, 3, System.Data.ParameterDirection.Input, true, ((System.Byte)(0)), ((System.Byte)(0)), "", System.Data.DataRowVersion.Current, "50"));
      this.procedureCommand.Parameters.Add(new Devart.Data.PostgreSql.PgSqlParameter("@pdname", Devart.Data.PostgreSql.PgSqlType.VarChar, 10, System.Data.ParameterDirection.Input, true, ((System.Byte)(0)), ((System.Byte)(0)), "", System.Data.DataRowVersion.Current, "RESEARCH"));
      this.procedureCommand.Parameters.Add(new Devart.Data.PostgreSql.PgSqlParameter("@ploc", Devart.Data.PostgreSql.PgSqlType.VarChar, 10, System.Data.ParameterDirection.Input, true, ((System.Byte)(0)), ((System.Byte)(0)), "", System.Data.DataRowVersion.Current, "NEW YORK"));
      this.procedureCommand.Owner = this;
      // 
      // selectCommand
      // 
      this.selectCommand.CommandText = "SELECT * FROM DEPT";
      this.selectCommand.Name = "selectCommand";
      this.selectCommand.Owner = this;
      // 
      // dataAdapter
      // 
      this.dataAdapter.SelectCommand = this.selectCommand;
      // 
      // dataGrid
      // 
      this.dataGrid.BorderStyle = System.Windows.Forms.BorderStyle.None;
      this.dataGrid.CaptionVisible = false;
      this.dataGrid.DataMember = "";
      this.dataGrid.Dock = System.Windows.Forms.DockStyle.Fill;
      this.dataGrid.HeaderForeColor = System.Drawing.SystemColors.ControlText;
      this.dataGrid.Location = new System.Drawing.Point(0, 88);
      this.dataGrid.Name = "dataGrid";
      this.dataGrid.Size = new System.Drawing.Size(477, 246);
      this.dataGrid.TabIndex = 10;
      // 
      // topPanel
      // 
      this.topPanel.Controls.AddRange(new System.Windows.Forms.Control[] {
                                                                           this.lbParams,
                                                                           this.tbLoc,
                                                                           this.tbDname,
                                                                           this.tbDeptno,
                                                                           this.lbParam3,
                                                                           this.lbParam2,
                                                                           this.lbParam1,
                                                                           this.btClear,
                                                                           this.btFill,
                                                                           this.btExecute,
                                                                           this.edStoredProcName,
                                                                           this.lbStoredProc});
      this.topPanel.Dock = System.Windows.Forms.DockStyle.Top;
      this.topPanel.Name = "topPanel";
      this.topPanel.Size = new System.Drawing.Size(477, 88);
      this.topPanel.TabIndex = 8;
      // 
      // lbParams
      // 
      this.lbParams.AutoSize = true;
      this.lbParams.Location = new System.Drawing.Point(8, 32);
      this.lbParams.Name = "lbParams";
      this.lbParams.Size = new System.Drawing.Size(155, 13);
      this.lbParams.TabIndex = 18;
      this.lbParams.Text = "Stored procedure parameters:";
      // 
      // tbLoc
      // 
      this.tbLoc.Location = new System.Drawing.Point(338, 52);
      this.tbLoc.Name = "tbLoc";
      this.tbLoc.Size = new System.Drawing.Size(127, 20);
      this.tbLoc.TabIndex = 17;
      this.tbLoc.Text = "";
      // 
      // tbDname
      // 
      this.tbDname.Location = new System.Drawing.Point(171, 52);
      this.tbDname.Name = "tbDname";
      this.tbDname.Size = new System.Drawing.Size(119, 20);
      this.tbDname.TabIndex = 16;
      this.tbDname.Text = "";
      // 
      // tbDeptno
      // 
      this.tbDeptno.Location = new System.Drawing.Point(47, 52);
      this.tbDeptno.Name = "tbDeptno";
      this.tbDeptno.Size = new System.Drawing.Size(65, 20);
      this.tbDeptno.TabIndex = 15;
      this.tbDeptno.Text = "";
      // 
      // lbParam3
      // 
      this.lbParam3.AutoSize = true;
      this.lbParam3.Location = new System.Drawing.Point(311, 56);
      this.lbParam3.Name = "lbParam3";
      this.lbParam3.Size = new System.Drawing.Size(22, 13);
      this.lbParam3.TabIndex = 14;
      this.lbParam3.Text = "Loc";
      // 
      // lbParam2
      // 
      this.lbParam2.AutoSize = true;
      this.lbParam2.Location = new System.Drawing.Point(128, 55);
      this.lbParam2.Name = "lbParam2";
      this.lbParam2.Size = new System.Drawing.Size(41, 13);
      this.lbParam2.TabIndex = 13;
      this.lbParam2.Text = "Dname";
      // 
      // lbParam1
      // 
      this.lbParam1.AutoSize = true;
      this.lbParam1.Location = new System.Drawing.Point(3, 54);
      this.lbParam1.Name = "lbParam1";
      this.lbParam1.Size = new System.Drawing.Size(41, 13);
      this.lbParam1.TabIndex = 12;
      this.lbParam1.Text = "Deptno";
      // 
      // btClear
      // 
      this.btClear.FlatStyle = System.Windows.Forms.FlatStyle.System;
      this.btClear.Location = new System.Drawing.Point(152, 0);
      this.btClear.Name = "btClear";
      this.btClear.TabIndex = 11;
      this.btClear.Text = "Clear";
      this.btClear.Click += new System.EventHandler(this.btClear_Click);
      // 
      // btFill
      // 
      this.btFill.FlatStyle = System.Windows.Forms.FlatStyle.System;
      this.btFill.Location = new System.Drawing.Point(76, 0);
      this.btFill.Name = "btFill";
      this.btFill.TabIndex = 10;
      this.btFill.Text = "Fill";
      this.btFill.Click += new System.EventHandler(this.btFill_Click);
      // 
      // btExecute
      // 
      this.btExecute.FlatStyle = System.Windows.Forms.FlatStyle.System;
      this.btExecute.Name = "btExecute";
      this.btExecute.TabIndex = 9;
      this.btExecute.Text = "Execute";
      this.btExecute.Click += new System.EventHandler(this.btExecute_Click);
      // 
      // edStoredProcName
      // 
      this.edStoredProcName.DropDownStyle = System.Windows.Forms.ComboBoxStyle.DropDownList;
      this.edStoredProcName.Items.AddRange(new object[] {
                                                          "Dept_Insert",
                                                          "Dept_Update"});
      this.edStoredProcName.Location = new System.Drawing.Point(328, 6);
      this.edStoredProcName.Name = "edStoredProcName";
      this.edStoredProcName.Size = new System.Drawing.Size(144, 21);
      this.edStoredProcName.TabIndex = 8;
      // 
      // lbStoredProc
      // 
      this.lbStoredProc.AutoSize = true;
      this.lbStoredProc.Location = new System.Drawing.Point(232, 8);
      this.lbStoredProc.Name = "lbStoredProc";
      this.lbStoredProc.Size = new System.Drawing.Size(92, 13);
      this.lbStoredProc.TabIndex = 7;
      this.lbStoredProc.Text = "Stored procedure";
      // 
      // StoredProcDemoControl
      // 
      this.Controls.AddRange(new System.Windows.Forms.Control[] {
                                                                  this.dataGrid,
                                                                  this.topPanel});
      this.Name = "StoredProcDemoControl";
      this.Size = new System.Drawing.Size(477, 334);
      ((System.ComponentModel.ISupportInitialize)(this.dataGrid)).EndInit();
      this.topPanel.ResumeLayout(false);
      this.ResumeLayout(false);

    }
    #endregion

    private void btExecute_Click(object sender, System.EventArgs e) {

      procedureCommand.CommandText = edStoredProcName.Text;
      procedureCommand.Parameters["@pdeptno"].Value = Convert.ToInt32(tbDeptno.Text);
      procedureCommand.Parameters["@pdname"].Value = tbDname.Text;
      procedureCommand.Parameters["@ploc"].Value = tbLoc.Text;
      procedureCommand.ExecuteNonQuery();
      writeStatus1 = "Procedure executed successfully";
      OnWriteStatus();
    }

    private void btFill_Click(object sender, System.EventArgs e) {

      System.Data.DataSet ds = new System.Data.DataSet();
      dataAdapter.Fill(ds);
      dataGrid.DataSource = ds.Tables[0];
    }

    private void btClear_Click(object sender, System.EventArgs e) {

      dataGrid.DataSource = null;
      writeStatus1 = "";
      OnWriteStatus();
    }
  }
}

